dining/views.py

```diff
@@ -0,0 +1,348 @@
+from datetime import datetime, timedelta
+
+from dining.constants import map_to_pretty
+from django.db.models import Sum
+from django.http import HttpResponseBadRequest, HttpResponseForbidden, JsonResponse
+from legacy.models import Account, DiningBalance, DiningTransaction
+from options.models import Option, get_option, get_value
+from pytz import timezone
+from rest_framework.views import APIView
+
+from studentlife.utils import date_iso_eastern, get_new_start_end
+
+
+class Dashboard(APIView):
+    def get(self, request, format=None):
+        if not self.request.user.is_authenticated:
+            return HttpResponseForbidden()
+
+        pennid = request.user.username
+
+        try:
+            uid = Account.objects.get(pennid=pennid).id
+        except Account.DoesNotExist:
+            return HttpResponseBadRequest()
+
+        json = self.balance(uid)
+
+        start, end = self.get_semester_start_end()
+
+        eastern = timezone("US/Eastern")
+        start = start.replace(tzinfo=eastern, hour=0, minute=0, second=0, microsecond=0)
+        end = end.replace(tzinfo=eastern, hour=0, minute=0, second=0, microsecond=0)
+        json["start-of-semester"] = date_iso_eastern(start)
+        json["end-of-semester"] = date_iso_eastern(end)
+
+        json["cards"] = {"recent-transactions": self.recent_transactions_card(uid)}
+
+        average_balances = self.get_average_balances(uid, start)
+
+        if average_balances:
```

Comment:
```suggestion
        if average_balances is not None:
```
Generally, relying on "falsiness" of return values isn't the best idea. It's a bit better to explicitly say what you're expecting in the "None" case, whether that's False, an empty list or the value None.